### ISSUE TITLE
Limitation of GRE

### ISSUE THREAD

**Issue by sturfee-petrl** [NVIDIA/gpu-rest-engine#17]
related to #16

# TLDR

1. Is current version GRE from the box works only for Caffe and TensorRT?
2. A not thread safe *gpu library* can't be launched under GRE? Like OpenGL.
3. GRE can be used only if *gpu library* allow explicitly bind a GPU physical device?

# Summary

I didn't found detailed description of what is GRE. Name is ***GPU** rest engine*. But really shared code was only in **comon.h**. Examples for Caffe and TensorRT can be used only for Caffe and TensorRT :simple_smile: .

I should implement a vertical scale for our **OpenGL+cuda** server. **OpenGL** is **not thread safe** and it can't bind a physical GPU device explicitly. I can't use even distribution system from **comon.h**. I am trying to proof to my team that it's not possible to use any code from GRE for this.

Thank you! Don't hesitate to ask me details

**Comment by flx42** [comment]
Again, sorry for the delay.

I will clarify the current status. It's a demo of inference.
